packages/plugins/experimental/plugin-transcription/src/components/Transcript/Transcript.tsx

```diff
@@ -169,70 +140,78 @@ export const Transcript: FC<TranscriptProps> = ({ attendableId, ignoreAttention,
+ */
+const createRowMeta = (rows: TranscriptRow[]): DxGridAxisMeta => ({
+  grid: rows.reduce<DxGridAxisMeta['grid']>((acc, item, rowIndex) => {
+    const size = item.size;
+    acc[rowIndex] = { size };
+    return acc;
+  }, {}),
+});
+
+/**
+ * Compute flattened rows from blocks.
+ */
+const useRows = (blocks?: TranscriptBlock[]): TranscriptRow[] => {
+  const [rows, setRows] = useState<TranscriptRow[]>([]);
+  useEffect(() => {
+    for (const block of blocks ?? []) {
+      const { author, segments } = block;
+      rows.push({ blockId: block.id, author, size: lineHeight });
+      for (const segment of segments) {
+        const { started, text } = segment;
+        rows.push({ blockId: block.id, ts: started, text: text.trim(), size: lineHeight });
       }
-      blockIndex++;
     }
+    setRows(rows);
+  }, [blocks]);
 
-    log('render', { duration: Date.now() - ts });
-    return rows;
-
-    // return blocks.flatMap((block, blockIndex) => {
-    //   return [
-    //     [blockIndex, -1, lineHeight + cellSpacing],
-    //     ...block.segments.map((segment, segmentIndex) => {
-    //       div.innerHTML = segment.text;
-    //       const height = measureSegmentSync(div, segment.text) + cellSpacing;
-    //       return [blockIndex, segmentIndex, height] as [number, number, number];
-    //     }),
-    //   ];
-    // });
-  }
+  return rows;
 };
 
-// TODO(burdon): Occasional random measurement error (-1 actual lines).
-// const measureSegmentSync = (div: HTMLDivElement, text: string): number => {
-//   div.innerHTML = text;
-//   void div.offsetHeight;
-//   return div.offsetHeight;
-// };
-
-const measureSegment = async (div: HTMLDivElement, text: string, estimate = false): Promise<number> => {
-  return new Promise((resolve) => {
-    div.innerHTML = text;
-    if (estimate) {
-      const { height } = div.getBoundingClientRect();
-      resolve(height);
-    } else {
-      requestAnimationFrame(() => {
-        const { height } = div.getBoundingClientRect();
-        resolve(height);
-      });
+/**
+ * Render rows offscreen and measure heights.
+ */
+const Measuring = ({
+  rows,
+  width,
+  onUpdate,
+}: {
+  rows: TranscriptRow[];
+  width: number;
+  onUpdate: (rows: TranscriptRow[]) => void;
+}) => {
+  const ref = useRef<HTMLDivElement>(null);
+  useLayoutEffect(() => {
+    const div = ref.current;
+    if (div) {
+      for (let i = 0; i < div.children.length; i++) {
+        const row: HTMLDivElement = div.children[i].children[0] as HTMLDivElement;
+        rows[i].size = row.offsetHeight + cellSpacing;
+      }
+
+      onUpdate(rows);
     }
-  });
-};
+  }, [rows]);
 
-export const formatElapsed = (start: Date, end: Date) => {
-  const { hours, minutes, seconds } = intervalToDuration({ start, end });
-  const pad = (n = 0) => String(n).padStart(2, '0');
-  return `${pad(hours)}:${pad(minutes)}:${pad(seconds)}`;
+  return (
+    <div ref={ref} className='absolute top-0 z-10 invisible'>
```

Comment:
```suggestion
    <div ref={ref} className='fixed top-full invisible' inert='' aria-hidden='true'>
```
This will ensure the browser definitely ignores this.

Reply:
what do you mean -- are you saying that this shouldn't work? (it seems to).

Reply:
If it works, it’s probably fine to leave it for now, though maybe leave a TODO about ensuring this never receives focus and definitely won’t occlude canonical content.

e.g. `z-10` would make this appear on top of other things, and `absolute` could cause this to become visible if its offset parent happens to have a range of positions, also without `inert` this could receive focus if it has focusable content.